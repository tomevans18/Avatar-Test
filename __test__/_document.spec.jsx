import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Document from '../pages/_document';

describe('Document', () => {
  let wrapper;
  let EnhancedApp;

  beforeEach(async () => {
    const props = await Document.getInitialProps({
      renderPage: (...args) => {
        EnhancedApp = args[0].enhanceApp;
        return args;
      },
    });
    wrapper = shallow(<Document {...props} />);
  });

  it('should render as expected', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should create correct enhanced app', () => {
    wrapper = shallow(<EnhancedApp />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
