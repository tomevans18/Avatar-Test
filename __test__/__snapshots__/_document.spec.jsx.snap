// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Document should create correct enhanced app 1`] = `<Component />`;

exports[`Document should render as expected 1`] = `
<html
  lang="en"
>
  <Head>
    <link
      href="/static/favicon.ico"
      rel="icon"
      type="image/x-icon"
    />
    <meta
      content="width=device-width, initial-scale=1.0, viewport-fit=cover"
      name="viewport"
    />
    <meta
      content="This is a meta description for this website!!"
      name="description"
    />
    <link
      href="/static/manifest.webmanifest"
      rel="manifest"
    />
  </Head>
  <body>
    <Main />
    <NextScript />
  </body>
</html>
`;
